src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -244,27 +231,21 @@ class BlockImporter(
               broadcastBlocks(blocks, weights)
               updateTxPool(importedBlocksData.map(_.block), Seq.empty)
               supervisor ! ProgressProtocol.ImportedBlock(block.number, internally)
-            case BlockEnqueued => ()
-            case DuplicateBlock => ()
-            case UnknownParent => () // This is normal when receiving broadcast blocks
             case ChainReorganised(oldBranch, newBranch, weights) =>
               updateTxPool(newBranch, oldBranch)
               broadcastBlocks(newBranch, weights)
-              newBranch.lastOption match {
-                case Some(newBlock) =>
-                  supervisor ! ProgressProtocol.ImportedBlock(newBlock.number, internally)
-                case None => ()
-              }
+              newBranch.lastOption.foreach(block =>
+                supervisor ! ProgressProtocol.ImportedBlock(block.number, internally)
+              )
             case BlockImportFailedDueToMissingNode(missingNodeException) if syncConfig.redownloadMissingStateNodes =>
               // state node re-download will be handled when downloading headers
               doLog(importMessages.missingStateNode(missingNodeException))
               Running
             case BlockImportFailedDueToMissingNode(missingNodeException) =>
               Task.raiseError(missingNodeException)
-            case BlockImportFailed(error) =>
-              if (informFetcherOnFail) {
-                fetcher ! BlockFetcher.BlockImportFailed(block.number, BlacklistReason.BlockImportError(error))
-              }
+            case BlockImportFailed(error) if informFetcherOnFail =>
+              fetcher ! BlockFetcher.BlockImportFailed(block.number, BlacklistReason.BlockImportError(error))
+            case _ => ()
```

Comment:
I think the previous implementation was done this way intentionally to get the exhaustive check. It's probably not a big deal as if we ever change `BlockImportResult` we will probably check here also anyway. Otherwise we can compromise with `case BlockEnqueued | DuplicateBlock | UnknownParent | BlockImportFailed(_)`.

Reply:
yeah, explicit cases are the way to go but I think we have to have the catch all in the end too